compiler/src/iree/compiler/Codegen/Transforms/Transforms.cpp

```diff
@@ -473,5 +475,122 @@ void populateRemoveDeadMemAllocPatterns(RewritePatternSet &patterns) {
   patterns.insert<RemoveDeadInterfaceBindings>(patterns.getContext());
 }
 
+void analyseAllocsForPacking(func::FuncOp funcOp, ArrayRef<Operation *> allocs,
+                             SmallVector<AliasGroup> &aliasGroups) {
+  // Represent of a group of allocations with overlapping liverange and the
+  // liveness of the overall group.
+  struct AllocGroup {
+    SmallVector<Operation *> allocs;
+    // Keep track of every operation where any of the alloc in the group is
+    // live.
+    // Liveness is represent as a set of Operations where the alloc is alive.
+    // To make it merge liveranges and check if a given Operation interfers
+    // with the liverange we store it as a DesneSet.
+    llvm::DenseSet<Operation *> liveness;
```

Comment:
Representing liveness using all set of operations within the live range seems strange to me. Ideally liveness can be represented by the ordered pair `[{BlockId, instruction}, {BlockId, instruction})` where the first block dominates the second block... If you dont plan to make the change now, please leave a comment stating that this is an overkill and could be made better if needed later on.

Reply:
More information of kind of representation that would work well here is https://dl.acm.org/doi/pdf/10.1145/330249.330250

Reply:
Yes I know this is how you usually represent liveranges but I'm trying to re-use MLIR liveness analysis. Representing liveness with MLIR region is not trivial as there is no classic CFG representation.